Nested embedding and mutable reciever method build error.

<!-- Please make sure to run `v up` before reporting any issues as it may have already been fixed.
     It's also advisable to update all relevant modules using `v outdated` and `v install` -->

<!-- You can use `v doctor` to fill up the next fields -->
**V version: 0.2.4 3fa9128** 
**OS: ArchLinux** 

**What did you do?**
```v
pub struct Boundary{}

pub fn (b Boundary) contains(x int, y int) bool {
	return false
}

//--------------------------

pub struct Base{
	Boundary
}

pub fn (mut b Base) on_event(x int, y int){
	if b.Boundary.contains(x, y) {}
	if b.contains(x, y) {}
}

//--------------------------

pub struct ListBox{
	Base
}

pub fn (mut lb ListBox) on_event(x int, y int){    // 1) works if remove mut
	if lb.Base.Boundary.contains(x, y) {}
	if lb.contains(x, y) {}    // 2) works if comment out.
}
```

**What did you expect to see?**
No error.

**What did you see instead?**
```
==================
/tmp/v_1000/gui_bug.13634214105560811351.tmp.c:11100: error: pointer expected
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.
```

--- a/vlib/v/gen/c/fn.v
+++ b/vlib/v/gen/c/fn.v
@@ -1047,10 +1047,15 @@ fn (mut g Gen) method_call(node ast.CallExpr) {
 		} else {
 			g.expr(node.left)
 		}
-		for embed in node.from_embed_types {
+		for i, embed in node.from_embed_types {
 			embed_sym := g.table.sym(embed)
 			embed_name := embed_sym.embed_name()
-			if node.left_type.is_ptr() {
+			is_left_ptr := if i == 0 {
+				node.left_type.is_ptr()
+			} else {
+				node.from_embed_types[i - 1].is_ptr()
+			}
+			if is_left_ptr {
 				g.write('->')
 			} else {
 				g.write('.')

--- a/vlib/v/tests/struct_multi_embed_method_call_test.v
+++ b/vlib/v/tests/struct_multi_embed_method_call_test.v
@@ -1,0 +1,33 @@
+pub struct Boundary {}
+
+pub fn (b Boundary) contains(x int, y int) bool {
+	return false
+}
+
+pub struct Base {
+	Boundary
+}
+
+pub fn (mut b Base) on_event(x int, y int) {
+	if b.Boundary.contains(x, y) {
+	}
+	if b.contains(x, y) {
+	}
+}
+
+pub struct ListBox {
+	Base
+}
+
+pub fn (mut lb ListBox) on_event(x int, y int) {
+	if lb.Base.Boundary.contains(x, y) {
+	}
+	if lb.contains(x, y) {
+	}
+}
+
+fn test_struct_multi_embed_method_call() {
+	mut list_box := ListBox{}
+	list_box.on_event(11, 22)
+	assert true
+}

